sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java

```diff
@@ -112,50 +255,349 @@ public EditContactSteps(WebDriverHelpers webDriverHelpers) {
+    webDriverHelpers.clickWebElementByText(PULMONARY_OPTIONS, text);
+  }
+
+  public void selectPreexistingConditionRenal(String text) {
+    webDriverHelpers.clickWebElementByText(RENAL_OPTIONS, text);
+  }
+
+  public void selectPreexistingConditionNeurologic(String text) {
+    webDriverHelpers.clickWebElementByText(NEUROLOGIC_OPTIONS, text);
+  }
+
+  public void selectPreexistingConditionCardiovascular(String text) {
+    webDriverHelpers.clickWebElementByText(CARDIOVASCULAR_OPTIONS, text);
+  }
+
+  public void fillAdditionalRelevantPreexistingConditions(String source) {
+    webDriverHelpers.clearAndFillInWebElement(ADDITIONAL_RELEVANT_PRE_CONDITIONS_TEXT, source);
+  }
+
+  public void selectVaccinationStatusForThisDisease(String vaccine) {
+    webDriverHelpers.selectFromCombobox(VACCINATION_STATUS_COMBOBOX, vaccine);
+  }
+
+  public void selectImmunosuppressiveTherapy(String text) {
+    webDriverHelpers.clickWebElementByText(IMMUNOSUPPRESSIVE_THERAPY_OPTIONS, text);
+  }
+
+  public void selectActiveInCare(String text) {
+    webDriverHelpers.clickWebElementByText(CARE_OVER_60_OPTIONS, text);
+  }
+
+  public void clickCancelFollowUpButton() {
+    if (editedContact.isCancelFollowUp()) {
+      webDriverHelpers.clickOnWebElementBySelector(CANCEL_FOLLOW_UP_BUTTON);
+    }
+  }
+
+  public void selectOverwriteFollowUp(String text) {
+    webDriverHelpers.clickWebElementByText(OVERWRITE_FOLLOW_UP_LABEL, text);
+  }
+
+  public void fillDateOfFollowUpUntil(LocalDate date) {
+    webDriverHelpers.clearAndFillInWebElement(FOLLOW_UP_UNTIL_DATE, formatter.format(date));
+  }
+
+  public void fillFollowUpStatusComment(String source) {
+    webDriverHelpers.clearAndFillInWebElement(FOLLOW_UP_STATUS_TEXT, source);
+  }
+
+  public void selectResponsibleContactOfficer(String vaccine) {
+    webDriverHelpers.selectFromCombobox(RESPONSIBLE_STATUS_OFFICER_COMBOBOX, vaccine);
+  }
+
+  public void fillGeneralComment(String source) {
+    webDriverHelpers.clearAndFillInWebElement(GENERAL_COMMENT, source);
   }
 
   public Contact collectContactData() {
+    Contact contactInfo = getContactInformation();
+    String classification =
+        webDriverHelpers.getCheckedOptionFromHorizontalOptionGroup(CONTACT_CLASSIFICATION);
     String collectedDateOfReport = webDriverHelpers.getValueFromWebElement(REPORT_DATE);
     LocalDate parsedDateOfReport = LocalDate.parse(collectedDateOfReport, formatter);
-    String collectedLastDateOfContact =
-        webDriverHelpers.getValueFromWebElement(DATE_OF_LAST_CONTACT_INPUT);
+    LocalDate parsedDateOfFirst =
+        (webDriverHelpers.isElementVisibleWithTimeout(FIRST_DAY_CONTACT_DATE, 1))
+            ? LocalDate.parse(
+                webDriverHelpers.getValueFromWebElement(FIRST_DAY_CONTACT_DATE), formatter)
+            : null;
+    // LocalDate parsedDateOfFollowUp =
+    //    LocalDate.parse(webDriverHelpers.getValueFromWebElement(FOLLOW_UP_UNTIL_DATE), formatter);
```

Comment:
this is commented because if letting it available the test would fail. not sure if it's a bug. needs to find out if it's an issue or a feature.

Reply:
Please pull the last changes from development, it was fixed. Do not commit commented code if there is not an existent bug. Please check now if it is working.

Reply:
added https://github.com/hzi-braunschweig/SORMAS-Project/issues/5634

Reply:
Please add the ticket after the comment.